app/tests/e2e/test_revocation.py

```diff
@@ -0,0 +1,249 @@
+
+        assert rev_record["state"] == "revoked"
+
+
+@pytest.mark.anyio
+async def test_publish_all_revocations_no_payload(
+    faber_client: RichAsyncClient, issue_revocable_credentials_to_alice_and_revoke: list
+):
+
+    await faber_client.post(
+        f"{CREDENTIALS_BASE_PATH}/publish-revocations",
+        json={"revocation_registry_credential_map": {}},
+    )
+
+    for cred in issue_revocable_credentials_to_alice_and_revoke:
+
+        rev_record = (
+            await faber_client.get(
+                f"{CREDENTIALS_BASE_PATH}/revocation/record"
+                + "?credential_exchange_id="
+                + cred["credential_id"][3:]
+            )
+        ).json()
+
+        assert rev_record["state"] == "revoked"
+
+
+@pytest.mark.anyio
+async def test_publish_one_revocation(
+    faber_client: RichAsyncClient, issue_revocable_credentials_to_alice_and_revoke: list
+):
+
+    response = (
+        await faber_client.get(
+            f"{CREDENTIALS_BASE_PATH}/revocation/record"
+            + "?credential_exchange_id="
+            + issue_revocable_credentials_to_alice_and_revoke[0]["credential_id"][3:]
+        )
+    ).json()
+
+    rev_reg_id = response["rev_reg_id"]
+
+    await faber_client.post(
+        f"{CREDENTIALS_BASE_PATH}/publish-revocations",
+        json={"revocation_registry_credential_map": {rev_reg_id: ["1"]}},
+    )
+
+    for cred in issue_revocable_credentials_to_alice_and_revoke:
+
+        rev_record = (
+            await faber_client.get(
+                f"{CREDENTIALS_BASE_PATH}/revocation/record"
+                + "?credential_exchange_id="
+                + cred["credential_id"][3:]
+            )
+        ).json()
+        print(rev_record)
+        if rev_record["cred_rev_id"] == "1":
+            assert rev_record["state"] == "revoked"
+        else:
+            assert rev_record["state"] == "issued"
+
+
+@pytest.mark.anyio
+async def test_publish_revocations_bad_payload(
+    faber_client: RichAsyncClient,
+):
+    with pytest.raises(HTTPException) as exc:
+
+        await faber_client.post(
+            f"{CREDENTIALS_BASE_PATH}/publish-revocations",
+            json={"revocation_registry_credential_map": "bad"},
+        )
+
+    assert_that(exc.value.status_code).is_equal_to(422)
+
+    with pytest.raises(HTTPException) as exc:
+
+        await faber_client.post(
+            f"{CREDENTIALS_BASE_PATH}/publish-revocations",
+            json={"revocation_registry_credential_map": {"bad": "bad"}},
+        )
+
+    assert_that(exc.value.status_code).is_equal_to(422)
+
+    with pytest.raises(HTTPException) as exc:
+
+        await faber_client.post(
+            f"{CREDENTIALS_BASE_PATH}/publish-revocations",
+            json={
+                "revocation_registry_credential_map": {
+                    "WgWxqztrNooG92RXvxSTWv:4:WgWxqztrNooG92RXvxSTWv:3:CL:20:tag:CL_ACCUM:0": [
+                        "bad"
+                    ]
+                }
+            },
+        )
+
```

Comment:
Nice tests! Similar for this one, just one more 404 test will be good -- one for bad rrid, one for bad cred ex id

Reply:
I found a function that returns the pending `cred_rev_id`s for a given `rev_reg_id` (I missed it before :clown_face: ). I'm gonna update die validate function